packages/wrangler/src/config/validation.ts

```diff
@@ -384,6 +396,26 @@ function normalizeAndValidateBaseDirField(
 	}
 }
 
+/**
+ * Validate the `pages_build_output_dir` field and return the normalized values.
+ */
+function normalizeAndValidatePagesBuildOutputDir(
+	configPath: string | undefined,
+	rawPagesDir: string | undefined
+): string | undefined {
+	const configDir = path.dirname(configPath ?? "wrangler.toml");
+	if (rawPagesDir !== undefined) {
+		if (typeof rawPagesDir === "string") {
+			const directory = path.resolve(configDir);
+			return path.resolve(directory, rawPagesDir);
+		} else {
+			return rawPagesDir;
```

Comment:
When could this branch occur?

Reply:
Ah I see you copied this from my code for `main` 😄 
I think that code is also overzealous here.